extensions/2.0/Khronos/KHR_materials_iridescence/README.md

```diff
@@ -0,0 +1,207 @@
+
+The values for iridescence intensity can be defined using a factor, a texture, or both.
+`iridescenceFactor` is multiplied with the red channel of `iridescenceTexture` to control the overall strength of the iridescence effect. If the texture is not set, a value of 1.0 is assumed for the texture. 
+```
+iridescence = iridescenceFactor * iridescenceTexture.r
+```
+
+If `iridescenceFactor` is zero (default), the iridescence extension has no effect on the material. 
+All textures in this extension use a single channel in linear space. 
+The thickness of the thin-film is set to `iridescenceThicknessMaximum` if `iridescenceThicknessTexture` is not given.
+If `iridescenceThicknessTexture` is set, the thickness of the thin-film varies between  `iridescenceThicknessMinimum` and `iridescenceThicknessMaximum` as follows: 
+```
+thickness = mix(iridescenceThicknessMinimum, iridescenceThicknessMaximum, iridescenceThicknessTexture.g)
+```
+
+Aside from light direction and IOR, the thickness of the thin-film defines the variation in hue. 
+This effect is the result of constructive and destructive interferences of certain wavelengths. 
+If the the optical path difference between the ray reflected at the thin-film and the ray reflected at the base material is half the wavelength (λ), the resulting 180 degree phase shift is cancelling out the reflected light:
+
+<figure>
+<img width=70% src="./figures/interference.png"/>
+<figcaption><em> </em></figcaption>
+</figure>
+
+With a thin-film thickness near half the wavelength of visible light (380 nm - 750 nm), the effect is most visible. 
+By increasing the thin-film thickness, multiples of wavelengths are still causing wave interferences, however, as the optical path distance increases, different rays are mixed in. 
+This leads to more pastel colored patterns for increased thickness:
+
+<figure>
+<img src="./figures/thickness-comparison.jpg"/>
+<figcaption><em>Comparison of different iridescence thickness ranges for a constant iridescence IOR value of 1.8. </em></figcaption>
+</figure>
+
+The thin-film layer can have a different IOR than the underlying material. With `iridescenceIOR` one can set an IOR value for the thin-film layer independently. 
+This also has an effect on the optical path difference:
+
+<figure>
+<img src="./figures/ior-comparison.jpg"/>
+<figcaption><em>Comparison of different iridescence IOR values for a thin-film thickness range between 200 nm (bottom) and 800 nm (top).</em></figcaption>
+</figure>
+
+
+The iridescence effect is modeled via a microfacet BRDF with a modified Fresnel reflectance term that accounts for inter-reflections as shown in [Laurent Belcour and Pascal Barla (2017)](https://belcour.github.io/blog/research/publication/2017/05/01/brdf-thin-film.html).
+
+
+## Implementation Notes
+
+*This section is non-normative.*
+
+For the calculation of `f_specular`, the original Fresnel term `F_schlick` is exchanged with a newly introduced `F_iridescence` term. 
```

Comment:
The term `F_schlick` does not exist in the base spec.

Reply:
@proog128 I updated the documentation quite a bit and added some improved implementation details

Reply:
Thanks!

This made me notice a tiny issue here:

```
iridescence_fresnel_lum = luminance(iridescence_fresnel)
return (1 - iridescene_fresnel_lum) * base + iridescence_fresnel * specular_brdf
```

In many cases this is not energy conserving (per RGB channel). To make it energy conserving, you have to use the maximum:

```
iridescence_fresnel_max = max(iridescence_fresnel.r, iridescence_fresnel.g, iridescence_fresnel.b)
return (1 - iridescence_fresnel_max) * base + iridescence_fresnel * specular_brdf
```

Reply:
I also know, that this might not be energy conserving but might/should preserve the perceived luminance. We can also go with the maximum though, which would make the diffuse darker (especially in cases where one RGB component of the iridescence Fresnel is very high).

Reply:
The maximum RGB component of the iridescence Fresnel is used for weighting as discussed in the PBR TSG meeting.